### ISSUE TITLE
how to import shapefile 3d on potree and to make it as point cloud?

### ISSUE THREAD

**Issue by Cannonbow** [potree/potree#1189]
Hello i hope you are doing well ?

So i found this project potree which is really good, normally i use QGIS but only for .tif file but when i searched for point cloud was looking how to do that and i found potree but it is not easy to use that i checked examples like shapefiles.html but i dont know how to import my own shapefile 3D with point cloud using potree. i'm still looking for how to do that in websites also on youtube but nothing to guide me : 
![image](https://user-images.githubusercontent.com/41770445/170234217-2c023a50-cb83-48e8-8cd7-1e1911b80d35.png)

i wanna make like this image i have shapefiles with buildings etc..... but i dont know how to make it look like it is in image in this example if you can help familiarize with potree it is more practical than using QGIS especially for measurements and more fonctionnalities

**Comment by tfwittwer** [comment]
If I understand correctly, you want to display the contents of a shapefile in the Potree viewer? Among the examples there's a shapefile.html that shows how to do this.

**Comment by nasrhaqiq** [comment]
> If I understand correctly, you want to display the contents of a shapefile in the Potree viewer? Among the examples there's a shapefile.html that shows how to do this.

the one im showing to you in this picture is shapefiles.html yes i wanna do the same like this so in potree i should have .las file or .shp file ? this is my question or we can do both ?

**Comment by tfwittwer** [comment]
Yes you can do both, display a point cloud (LAS converted to Potree format) and a shapefile as overlay. Have you been able to display the point cloud without shapefile?

**Comment by nasrhaqiq** [comment]
> Yes you can do both, display a point cloud (LAS converted to Potree format) and a shapefile as overlay. Have you been able to display the point cloud without shapefile?

to be honest no i never did that so i'm new to this i just wanna know how i can make a shapefile looks like that shapefiles.html, of course it is a point cloud so do we need drone to create a point cloud or you can do it without,

**Comment by tfwittwer** [comment]
Shapefiles and point clouds are two different things. Shapefiles are used for vector geometry, i.e. lines and polygons, and small numbers of points as well. You can create a shapefile from vector data using a GIS program, e.g. QGIS.
Point clouds are acquired through laser scanning or by processing stereographic imagery.

So if you want to create something that looks like the shapefiles.html example, you need a vector data set and a point cloud that cover the same area. But it is still unclear to me what exactly you're trying to achieve. Potree is mostly a viewer. If you are looking for a software to do more with point clouds, such as measuring, I would suggest CloudCompare.

**Comment by nasrhaqiq** [comment]
> Shapefiles and point clouds are two different things. Shapefiles are used for vector geometry, i.e. lines and polygons, and small numbers of points as well. You can create a shapefile from vector data using a GIS program, e.g. QGIS. Point clouds are acquired through laser scanning or by processing stereographic imagery.
> 
> So if you want to create something that looks like the shapefiles.html example, you need a vector data set and a point cloud that cover the same area. But it is still unclear to me what exactly you're trying to achieve. Potree is mostly a viewer. If you are looking for a software to do more with point clouds, such as measuring, I would suggest CloudCompare.

yes i want to create somethink like shapefiles.html, and yes i have shapefiles of my city (building,landuse,roads etc......) so i know potree is just a viewer so im looking for how to make point cloud i searched i found software called bentley contextcapture if CloudCompare do the same job and opensource i should use it then. What i'm trying to achieve i want to make point cloud of my City like in this example of shapefiles.html but it is my first experience. i have also drone DJI Phantom 4 RTK does it do the job if i capture images and change it to point cloud ? Thanks for helping me i really appreciate it

**Comment by tfwittwer** [comment]
ContextCapture is paid software. CloudCompare is not a software for creating point clouds from images, but for many other point cloud operations. Google "free photogrammetry software" for software that can do what you want. I think Meshroom is a popular option.

**Comment by nasrhaqiq** [comment]
> ContextCapture is paid software. CloudCompare is not a software for creating point clouds from images, but for many other point cloud operations. Google "free photogrammetry software" for software that can do what you want. I think Meshroom is a popular option.

Thank you for helping me so yeah ContextCapture is paid software, so i went to work with Agisoft Metashape and worked for me the point cloud and exported it to .las so i use potree desktop but after that i get this problem 
![image](https://user-images.githubusercontent.com/41770445/172743206-adc68a9e-811a-482c-b8de-805b2c17643e.png)

but with potree i see it like this 
![image](https://user-images.githubusercontent.com/41770445/172743375-69101b26-4b0c-405b-a44a-7ca8474dc864.png)

if you have any idea to make it visualised like in metashape.

The step i did : 
- i imported the images.
- i changed photos to point cloud 
- we see the point cloud in the 1st image i showed you.

**Comment by nasrhaqiq** [comment]
i think i found issue i should use local coordinates haha so here is result it is good but i dont know why it appear with big points and not small as it should be 
![image](https://user-images.githubusercontent.com/41770445/172744054-e1be847a-dfbc-4ec8-a586-3f4144eb4c10.png)

**Comment by tfwittwer** [comment]
You can set the point size by selecting the point cloud under _Scene_ . I think it uses this point size as your point cloud is quite sparse. With smaller points you would have large voids and look through the cloud.

**Comment by nasrhaqiq** [comment]
> You can set the point size by selecting the point cloud under _Scene_ . I think it uses this point size as your point cloud is quite sparse. With smaller points you would have large voids and look through the cloud.

sure bro but i dont know how to make it small points so it is doable in Potree viewer ? oh i see they are 2 types of point cloud they are sparse point cloud which is this and other one is dense point cloud to be honest i didnt work on dense should i make it dense with metashape ?

**Comment by tfwittwer** [comment]
If that is what you want, sure.

**Comment by nasrhaqiq** [comment]
> If that is what you want, sure.

yes i want it to look like in shapefiles.html so im doing build dense in metashape it takes too much times 2 - 3 hours hope it will work thank you so much for help i appreciate that <3

**Comment by nasrhaqiq** [comment]
@tfwittwer is there a way to make point cloud size smaller 1st time i'm working with metashape 
![image](https://user-images.githubusercontent.com/41770445/172866522-0b21aada-bfd1-4def-b62c-0ed73ff18d63.png)

like points in Shapefiles.html

**Comment by tfwittwer** [comment]
Sure, you can use CloudCompare for cutting it into smaller pieces and downsampling.

**Comment by nasrhaqiq** [comment]
> Sure, you can use CloudCompare for cutting it into smaller pieces and downsampling.

oh so CloudCompare do this job i see now thank you so much i will do that

**Comment by nasrhaqiq** [comment]
@tfwittwer sorry again so i'm with cloudcompare now and i imported my .las file 
![image](https://user-images.githubusercontent.com/41770445/172964119-8967f572-6d42-4cd5-9c66-4536e8c4bc92.png)

if you zoom it you see points not sticked to each other there is a space between them : 
![image](https://user-images.githubusercontent.com/41770445/172964293-995a36c4-aead-407e-b328-6efad089921c.png)

do you know how i can make it sticked to each other thank you so much it is my 1st time working with cloud compare

**Comment by tfwittwer** [comment]
Well, they are points and hence cover an infinitely small area. You can increase the point size by selecting the point cloud and then clicking on + in the upper left corner of the view screen (or changing the point size in the attributes window). If you want them to "stick together", you could compute a mesh, but then you no longer have a point cloud.

**Comment by nasrhaqiq** [comment]
> Well, they are points and hence cover an infinitely small area. You can increase the point size by selecting the point cloud and then clicking on + in the upper left corner of the view screen (or changing the point size in the attributes window). If you want them to "stick together", you could compute a mesh, but then you no longer have a point cloud.

I see so if i stick em together i need to make a mesh that shapefiles.html isnt a mesh right ? then i just work on point cloud size thats what i could do

**Comment by tfwittwer** [comment]
That's correct, the example shown in shapefiles.html is not a mesh. Potree does support displaying meshes in PLY or OBJ form, so that's an option if you would prefer a mesh.

**Comment by nasrhaqiq** [comment]
> That's correct, the example shown in shapefiles.html is not a mesh. Potree does support displaying meshes in PLY or OBJ form, so that's an option if you would prefer a mesh.

thank you so much i will try mesh and dense too and see which is the best one
